commit 6ad569611a1b5c7e4d8c2f019d04bacaa85dbbce
Author: Guillaume Nault <gnault@redhat.com>
Date:   Wed Oct 8 21:12:23 2025 +0200

    man/man7/rtnetlink.7: Document metrics attributes (RTAX_*).
    
    Add a brief explanation of the RTAX_* attributes that can be used in
    RTA_METRICS.
    
    Signed-off-by: Guillaume Nault <gnault@redhat.com>
    Message-ID: <67e7f52aad3d777ef1075eca902d0d235e0d3753.1759950455.git.gnault@redhat.com>
    [alx: ffix]
    Signed-off-by: Alejandro Colomar <alx@kernel.org>

    Conflicts:
        Fixed a missing dot typo.

diff -Nrup a/man7/rtnetlink.7 b/man7/rtnetlink.7
--- a/man7/rtnetlink.7	2024-02-11 19:59:38.000000000 -0500
+++ b/man7/rtnetlink.7	2025-11-19 11:05:46.395976625 -0500
@@ -329,10 +329,14 @@ RTA_OIF:int:Output interface index
 RTA_GATEWAY:protocol address:The gateway of the route
 RTA_PRIORITY:int:Priority of route
 RTA_PREFSRC:protocol address:Preferred source address
-RTA_METRICS:int:Route metric
+RTA_METRICS::T{
+Route metrics
+.br
+(see below).
+T}
 RTA_MULTIPATH::T{
 Multipath nexthop data
-br
+.br
 (see below).
 T}
 RTA_PROTOINFO::No longer used
@@ -371,6 +375,83 @@ routes (in seconds)
 T}
 .TE
 .IP
+.B RTA_METRICS
+contains an array of
+.I struct rtattr
+with their corresponding attributes:
+.RS
+.IP
+.TS
+tab(:);
+c s s
+lb l lx.
+Attributes
+rta_type:Value type:Description
+_
+RTAX_UNSPEC:-:unspecified
+RTAX_LOCK:__u32:T{
+Bit field indicating which RTAX_* attributes are locked.
+T}
+RTAX_MTU:__u32:T{
+Maximum Transmission Unit for this route.
+T}
+RTAX_WINDOW:__u32:T{
+Maximum size of the receive window for this route.
+T}
+RTAX_RTT:__u32:T{
+Estimated round-trip time for this route.
+T}
+RTAX_RTTVAR:__u32:T{
+Estimated round-trip time variation for this route.
+T}
+RTAX_SSTHRESH:__u32:T{
+Slow start threshold to use for this route.
+T}
+RTAX_CWND:__u32:T{
+Maximum size of the congestion window for this route.
+T}
+RTAX_ADVMSS:__u32:T{
+Maximum Segment Size to advertise for this route.
+T}
+RTAX_REORDERING:__u32:T{
+Initial reordering level of packets for this route.
+T}
+RTAX_HOPLIMIT:__u32:T{
+Hop limit (TTL) to use for this route.
+T}
+RTAX_INITCWND:__u32:T{
+Initial congestion window to use for this route.
+T}
+RTAX_FEATURES:__u32:T{
+Features to enable for this route specifically.
+T}
+RTAX_RTO_MIN:__u32:T{
+Minimum Retransmission TimeOut to use for this route.
+T}
+RTAX_INITRWND:__u32:T{
+Initial size of the receive window for this route.
+T}
+RTAX_QUICKACK:__u32:T{
+Use quick ack for this route.
+T}
+RTAX_CC_ALGO:asciiz string:T{
+Congestion Control algorithm to use for this route.
+T}
+RTAX_FASTOPEN_NO_COOKIE:__u32:T{
+Allow TCP Fast Open without cookie.
+T}
+.TE
+.RE
+.IP
+Metrics that are locked with
+.B RTAX_LOCK
+take precedence over the values normally used by the kernel
+(computed or assigned by a sysctl or setsockopt(2)).
+Therefore, some metrics, like
+.BR RTAX_RTO_MIN ,
+have no effect unless their bit is set in
+.BR RTAX_LOCK .
+.IP
 .B RTA_MULTIPATH
 contains several packed instances of
 .I struct rtnexthop
